newton/_src/geometry/terrain_generator.py

```
@@ -551,3 +600,109 @@ def _to_newton_mesh(vertices: np.ndarray, indices: np.ndarray) -> tuple[np.ndarr
+        extent_x: Physical size in X direction (meters)
+        extent_y: Physical size in Y direction (meters)
+        center_x: Center X coordinate (default: 0.0)
+        center_y: Center Y coordinate (default: 0.0)
+        ground_z: Z coordinate of bottom surface (default: 0.0)
+
+    Returns:
+        tuple of (vertices, indices) where vertices is (N, 3) float32 array
+        and indices is (M,) int32 array of triangle indices (flattened)
+    """
+    if heightfield.ndim != 2:
+        raise ValueError(f"heightfield must be 2D array, got shape {heightfield.shape}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)